openmicroscopy/openmicroscopy
# Allow customizing tree

This PR adds `map` node type to OMERO.web (at the moment new type is hardcoded) and allows handling custom attributes passed to the tree. Additional bug fixes

To test:
 - this PR is tested with https://github.com/aleksandra-tarkowska/mapannotations
 - `pip install git+git://github.com/aleksandra-tarkowska/mapannotations.git`

 - and:

 ```
/home/omero/workspace/OMERO-web/OMERO.web/bin/omero config append omero.web.apps '"mapr"'
/home/omero/workspace/OMERO-web/OMERO.web/bin/omero config set omero.web.ui.top_links '[["IDR", {"viewname": "webindex", "query_string": {"experimenter": -1}}, {"title": "Image Data Repository"}], ["Genes", {"viewname": "maprindex_gene", "query_string": {"experimenter": -1}}, {"title": "Genes browser"}], ["Phenotypes", {"viewname": "maprindex_phenotype", "query_string": {"experimenter": -1}}, {"title": "Phenotypes browser"}], ["siRNA", {"viewname": "maprindex_sirna", "query_string": {"experimenter": -1}}, {"title": "siRNA browser"}], ["Compound", {"viewname": "maprindex_compound", "query_string": {"experimenter": -1}}, {"title": "Compound browser"}], ["History", "history", {"title": "History"}], ["About", "/about/"], ["Help", "http://help.openmicroscopy.org/", {"title":"Open OMERO user guide in a new tab", "target":"new"}]]'

 ```

for full example see console output in https://10.0.51.150:8443/job/OMERO-web/

*This PR could be discussed in terms of how pagination will be handled in the main data tree when mixed types of nodes are loaded per page. At the moment nodes without counter are not using pagination*
